include/zephyr/drivers/i3c.h

```diff
@@ -1100,6 +1214,101 @@ static inline int i3c_recover_bus(const struct device *dev)
 	return api->recover_bus(dev);
 }
 
+/**
+ * @brief Attach an I3C device
+ *
+ * Called to attach a I3C device to the addresses. This is
+ * typicalled before a SETDASA or ENTDAA to reserve the addresses.
+ *
+ * @warning
+ * Use cases involving multiple writers to the i3c/i2c devices must prevent
+ * concurrent write operations, either by preventing all writers from
+ * being preempted or by using a mutex to govern writes to the i3c/i2c devices.
+ *
+ * @param target Pointer to the target device descriptor
+ *
+ * @retval 0 If successful.
+ * @retval -EINVAL If address is not available or if the device
+ *     has already been attached before
+ */
+static inline int i3c_attach_i3c_device(struct i3c_device_desc *target)
```

Comment:
Since this is marked `inline`, it will possibly be expanded at the caller site. Please add a helper function in C code for all this, and have this call the helper function instead. Ditto for the various additions below.

Reply:
I assume you're referring to just put the "figuring out the DA" decision tree in a helper function. Yeah, I'll do that

Reply:
I am thinking the whole function body. This type of stubs are supposed to be really light weight and is to call the driver API directly (if API func exists).

Reply:
ok...?? If I put the whole function within a function then there wouldn't be a point to the upper function still being around as you should just be calling the helper function which should always be implemented (and then it would call the driver api (if it exist)) :/ I understand your point about inline expanding a function that's a bit more than few lines, but doing that would just make the i3c api func unnecessary imo.

Would it make sense to remove these "inline" apis and just expect the application to call the helper functions?

Reply:
Oh yeah... just remove inline and leave it as a function prototype, and put implementation somewhere else.

(... my brain is not working today)

Reply:
done